### PR TITLE
merging beta branch, and doing appium beta cleanup tasks

### PR DESCRIPTION
**Pull request by Jonahss**

### CODE REVIEW

**Comment by Jonahss** [comment]
will version and publish after this PR

File: lib/xcode.js [hunk 1]
```diff
@@ -124,14 +122,14 @@ async function getVersionWithoutRetry () {
...
+const getVersionMemoized = _.memoize(
   function (retries = DEFAULT_NUMBER_OF_RETRIES) {
     return retry(retries, getVersionWithoutRetry);
   }
 );
 
 async function getVersion (parse = false, retries = DEFAULT_NUMBER_OF_RETRIES) {
-  let version = await getVersionMemorized(retries);
+  let version = await getVersionMemoized(retries);
   if (!parse) {
```
Comment (imurchie) [inline lib/xcode.js:131]: Since you've imported `parse` this local variable will clash now.

**Comment by Jonahss** [comment]
er. ci failed. fixing...

**Comment by jlipps** [comment]
looks like it complained because of mochawait. but good news because mochawait is no longer required!

File: lib/xcode.js [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
 import { retry } from 'asyncbox';
 import _ from 'lodash';
-import plist from 'plist';
+import { parse } from 'plist';
 import { exec } from 'teen_process';
-import bplistParser from 'bplist-parser';
-import B from 'bluebird';
 
-let parseFile = B.promisify(bplistParser.parseFile);
+let parsePlistData = parse;
```
Comment (imurchie) [inline lib/xcode.js]: Does `import { x as y} from z;` not work?

**Comment by Jonahss** [comment]
@jlipps yeah I removed mochawait as a dep, but I guess it needs to be removed from appium-gulp-plugins as well.

File: lib/xcode.js (hunk 2 above)
Comment (Jonahss) [inline lib/xcode.js]: ooh that's what I was looking for. Thanks.

**Comment by imurchie** [comment]
mochawait is still being imported in xcode-specs.

**Comment by Jonahss** [comment]
ah, that's it isaac thanks. Hoo boy, glad you said that, because this is what the error looks like on my machine:

```
[14:31:53] Error in plugin 'gulp-mocha'
Message:
    Cannot find module 'babel-runtime/helpers/sliced-to-array'
Details:
    code: MODULE_NOT_FOUND
Stack:
Error: Cannot find module 'babel-runtime/helpers/sliced-to-array'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:286:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/jonahss/Workspace/node_modules/mochawait/build/lib/mochawait.js:3:22)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/jonahss/Workspace/node_modules/mochawait/build/index.js:3:1)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (test/xcode-specs.js:5:46)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at /Users/jonahss/Workspace/appium-xcode/node_modules/appium-gulp-plugins/node_modules/mocha/lib/mocha.js:216:27
    at Array.forEach (native)
    at Mocha.loadFiles (/Users/jonahss/Workspace/appium-xcode/node_modules/appium-gulp-plugins/node_modules/mocha/lib/mocha.js:213:14)
```

**Comment by Jonahss** [comment]
but removing that import line fixes it. The error doesn't seem easy to figure out at all.

**Comment by imurchie** [comment]
Ah! Babelized stack traces.

**Comment by jlipps** [comment]
the CI error explicitly mentioned mochawait....

**Comment by Jonahss** [comment]
that's what we get for using the cool stuff ;)

**Pull request merged**